b+tree.ts

```diff
@@ -30,17 +30,91 @@ type index = number;
 //   - Objects can be used like arrays (e.g. have length property) but are slower
 //   - V8 source (NewElementsCapacity in src/objects.h): arrays grow by 50% + 16 elements
 
-/** Compares two numbers, strings, arrays of numbers/strings, Dates,
- *  or objects that have a valueOf() method returning a number or string. 
- *  Optimized for numbers. Returns 1 if a>b, -1 if a<b, and 0 if a===b.
+/**
+ * numbers, strings and arrays thereof, and objects that have a valueOf() method returning a number or string like Dates.
```

Comment:
I'd document this as "types that BTree supports by default"... assuming people can read types themselves.